pkg/daemon/server/service/rater/rater.go

```diff
@@ -217,15 +218,9 @@ func sleep(ctx context.Context, duration time.Duration) {
 // getPodReadCounts returns the total number of messages read by the pod
 // since a pod can read from multiple partitions, we will return a map of partition to read count.
 func (r *Rater) getPodReadCounts(vertexName, vertexType, podName string) *PodReadCount {
-	metricNames := map[string]string{
-		keyVertexTypeReduce: "reduce_isb_reader_data_read",
-		keyVertexTypeSource: "source_forwarder_read_total",
-		keyVertexTypeSink:   "sink_forwarder_data_read",
-	}
-
-	readTotalMetricName, ok := metricNames[vertexType]
-	if !ok {
-		readTotalMetricName = "forwarder_data_read"
+	readTotalMetricName := "forwarder_data_read_total"
+	if keyVertexTypeSource == vertexType {
+		readTotalMetricName = "forwarder_read_total"
```

Comment:
For source, we use `forwarder_read_total` instead of `forwarder_data_read_total`. Is it because currently source only processes real data messages hence the numbers are the same between these two metrics? If so I wonder if we can also use `forwarder_data_read_total` for source, since we plan to look into source idling and could introduce other non-data messages.

Reply:
> For source, we use `forwarder_read_total` instead of `forwarder_data_read_total`. Is it because currently source only processes real data messages hence the numbers are the same between these two metrics? If so I wonder if we can also use `forwarder_data_read_total` for source, since we plan to look into source idling and could introduce other non-data messages.

Wan't it you added this logic? 

There's no `data_read` metric exposed in the source.

Reply:
Yes, I added this logic. Now that we are unifying the metrics, I think we can change the source forwarder to also use `data_read`. There are two benefits: 1. it potentially paves the road for future source idling implementations and 2. with all vertices sharing the same metric name, we can completely remove the `vertexType` part of the raptor podkey, simplifying the code.